example-serverless-app-reuse/reuse-auth-only.yaml

```diff
@@ -4,15 +4,67 @@
 AWSTemplateFormatVersion: "2010-09-09"
 Transform: AWS::Serverless-2016-10-31
 Description: >
-  Example stack that shows how to reuse only the auth portion of the serverless application
-  in your own CloudFront distribution. Note that after deployment, you should manually update the
-  User Pool client, and point the Callback URL and Sign out URL at the CloudFront distribution. To work
-  around this circular dependency, you can also add an alternate domain name to the CloudFront distribution,
-  and provide that as parameter to the serverless stack. Then the User Pool client will be set to the right
-  Callback URL and Sign out URL automatically.
+  Example stack to that reuses the auth portion of the serverless application
+  in your own CloudFront distribution. 
 
-Resources:
+  After deployment, manually update the User Pool client to point the 'Callback URL' and 'Sign out URL' to the 
+  URL of the CloudFront distribution. 
+  
+  Alternately, add an alternate domain name to the CloudFront distribution and provide the name as parameter to the 
+  serverless stack. The alternate domain will also require adding a certificate to the Cloudfront Distribution.  
+  This alternative will set  the User Pool client to the right Callback URL and Sign out URL automatically.
+
+  The sample CloudFront distribution illustrates a full pattern of use -- modify it as needed.
+
+  The HttpHeaders parameter and corresponding reference in the 'LambdaEdgeProtection' nested application Resource
+  illustrates how to pass information to the underlying SAM app
+
+  Outputs illustrate retrieving values from the underlying SAM application
+
+Parameters: 
+  BucketNameParameter: 
+    Type: String
+    Description: A legal bucket name.  Must not exist.
+
+  HttpHeaders:
+    Type: String
+    Description: The HTTP headers to set on all responses from CloudFront. To be provided as a JSON object
+    Default: >-
+      {
+        "Content-Security-Policy": "default-src 'none'; img-src 'self'; script-src 'self'; style-src 'self'; object-src 'none'; connect-src 'self' https://*.amazonaws.com https://*.amazoncognito.com",
```

Comment:
Idea: let's do a really lenient CSP here? Because that is the first thing many people run in to when deploying their own web app

Reply:
Like the idea, but a quick dip into search suggests there isn't much agreement on what a lenient policy is that actually works.  If you've got one, do let me know.  Best I've seen is from [here](https://stackoverflow.com/questions/38461211/what-is-the-maximally-permissive-content-security-policy), and looks like `Content-Security-Policy-Report-Only: default-src 'none' `.  I'll give it a try, but there were reports it still blocked some bits and pieces.

Reply:
Maybe just do:
{}

And put in comment that this is a No-Op, useful for DEV only, that should be revisited.

Reply:
Could you please clarify where you see the {} inserted?  There are other headers in the default headers that have value and, I think, always need to be there

Reply:
For the HTTP Headers param use this Default:

```
    Default: "{}"  # Some explaining comment here
```

Or if you like it better make it whatever else, as long as it is different from the normal template.yaml there is a lesson it it I guess (the lesson is, you can easily put your own values in):

```
    Default: >-
      {
        "Strict-Transport-Security": "max-age=31536000; includeSubdomains; preload",
      }
```

Reply:
Interesting solution to use the report only CSP